src/auth/digest/Config.cc

```diff
@@ -826,11 +826,14 @@ Auth::Digest::Config::decode(char const *proxy_auth, const HttpRequest *request,
             break;
 
         case DIGEST_NC:
-            if (value.size() != 8) {
+            if (value.size() == 8) {
+                // For historical reasons, the nc value MUST be exactly 8 hexadecimal
+                // digits.
+                xstrncpy(digest_request->nc, value.rawBuf(), value.size() + 1);
```

Comment:
To explain why this terrible code "works" and protect it from refactoring bugs, I would add an assertion like this (untested) one:

```suggestion
                static_assert(sizeof(digest_request->nc) == 8 + 1);
                xstrncpy(digest_request->nc, value.rawBuf(), value.size() + 1);
```